arch/rv32i/src/syscall.rs

```diff
@@ -166,6 +166,16 @@ impl kernel::syscall::UserspaceKernelBoundary for SysCall {
         _app_brk: *const u8,
         state: &mut Riscv32iStoredState,
     ) -> (ContextSwitchReason, Option<*const u8>) {
+        // We need to ensure that all writes issued in the kernel are
+        // committed to memory prior to switching to userspace, in
+        // order to provide a consistent view on
+        // application-accessible buffers.
```

Comment:
```suggestion
        // We need to ensure that the compiler does not reorder
        // kernel memory writes to after the userspace context switch
        // to ensure we provide a consistent memory view of
        // application-accessible buffers.
```

I think this clarifies that this is about compiler instruction ordering and not pipelining in the microarchitecture.